ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -170,3 +159,17 @@ fetchDecisions
                       }
               traverse_ demoteCSJDynamoAndIgnoreInflightBlocks peersOrderCurrent
               pure peersOrder'
+
+      checkChangeOfCurrentPeer :: Maybe (any, PeerInfo header peer extra) -> PeersOrder peer -> m ()
+      checkChangeOfCurrentPeer theDecision PeersOrder {peersOrderCurrent, peersOrderOthers} =
```

Comment:
Maybe rename to `maybeSetCurrentPeer`.

Reply:
It does not just set the current peer, though, it also demotes the previous one. This is only done by calling a callback, though, so it's probably OK if it doesn't appear in the name. I'll go for `maybeSetCurrentPeerAndDemote` but let me know if you think this is too much.

Reply:
Done in 7044561f22d538f4bd2511dc08197b5ac5af0aa2.